laravel-app/app/Http/Controllers/ProfileController.php

```diff
@@ -64,23 +64,25 @@ public function show(int $userId): RedirectResponse|View
     {
         $user = User::with('posts.likes')->find($userId);
 
+        if ($user !== null) {
+            $posts = $user->posts()
+                ->orderBy('created_at', 'desc')
+                ->paginate(4);
+        }
+
         if (!$user) {
             $error = ['error' => 'No user with that id found.'];
             return redirect()->back()->withErrors($error);
         }
 
-        $likesCount = $user->posts->sum(function ($post) {
-            return $post->likes->count();
-        });
-
         /** @var User $user */
         $imagePath = $this->getImagePath($user);
         $authUser = auth()->user();
         return view('profile.show', [
             'authUser' => $authUser,
             'user' => $user,
-            'likesCount' => $likesCount,
             'imagePath' => $imagePath,
```

Comment:
use model for the path

Reply:
tranferred to model as attribute sir, thank you ❤️